hack/test-templates.sh

```diff
@@ -234,32 +258,37 @@ nginx_image="ghcr.io/stargz-containers/nginx:1.19-alpine-org"
 alpine_image="ghcr.io/containerd/alpine:3.14.0"
 
 if [[ -n ${CHECKS["container-engine"]} ]]; then
+	sudo=""
+	if [[ ${NAME} == "wsl2" ]]; then
```

Comment:
This should check the vmType, not the instance name

Reply:
Do you have a snippet what is the recommended way to get vmType from inside the shell script.

Also, I'm thinking that this might be extracted to a variable "rootfulContainerEngine" and set it per template. I genuinely believe that it is possible to fix WSL2 template to support user level nerdctl instead. Or there is a possibility that such WSL2 template could appear.

WDYT?

Reply:
> vmType

```bash
limactl ls --json "${NAME}" | jq -r .vmType
```

> rootfulContainerEngine

No need to introduce a new variable:
```bash
docker info --format json | jq -r '.SecurityOptions | if index("name=rootless") then "rootless" else "rootful" end'
```

Reply:
For now I opted for vmType check. Check for rootless will be added when it is really needed (when WSL2 will be able to run nerdctl as non-root).